### PR TITLE
Cleanup generate code attributes and enable SyleCop Analyzers

### PR DESCRIPTION
**Pull request by pepone**
See #1491

### CODE REVIEW

File: csharp/msbuild/CodeAnalysis.Base.globalconfig [hunk 1]
```diff
@@ -264,6 +291,102 @@ dotnet_diagnostic.CA1859.severity = none
 
 # TODO review rules below
```
Comment (pepone) [inline csharp/msbuild/CodeAnalysis.Base.globalconfig:293]: Most of the rules below must be enabled, and fix the warnings but we can do in separate PRs.

**Review by bernardnormier — APPROVED** [review]

**Review by externl — APPROVED** [review]
